packages/react/src/evm/providers/thirdweb-provider.tsx

```diff
@@ -80,6 +81,22 @@ export const ThirdwebProvider = <
     [signer],
   );
 
+  // preload the embeddedWallet SDK if present in supportedWallets
+  const ewsRef = useRef(false);
+  useEffect(() => {
+    if (ewsRef.current) {
+      return;
+    }
+    ewsRef.current = true;
+    const hasEmbeddedWallet = wallets.find((w) => w.id === "embeddedWallet");
+    if (hasEmbeddedWallet && restProps.clientId) {
+      new EmbeddedWalletSdk({
+        clientId: restProps.clientId,
+        chain: "Ethereum",
+      });
```

Comment:
Since the `EmbeddedWalletSdk` is not a dynamic import - it can not be tree-shaken. we should use a dynamic import of `EmbeddedWalletSdk` here

Reply:
done